src/exchanges/EngineAdapter.sol

```diff
@@ -1,71 +1,105 @@
 pragma solidity 0.6.1;
 pragma experimental ABIEncoderV2;
 
-import "../engine/Engine.sol";
-import "../fund/hub/Hub.sol";
-import "../fund/trading/Trading.sol";
-import "../fund/vault/Vault.sol";
-import "../dependencies/DSMath.sol";
-import "../dependencies/WETH.sol";
-import "../dependencies/token/IERC20.sol";
 import "./ExchangeAdapter.sol";
-import "../dependencies/TokenUser.sol";
+import "./OrderFiller.sol";
+import "../dependencies/WETH.sol";
+import "../engine/IEngine.sol";
 
+/// @title EngineAdapter Contract
+/// @author Melonport AG <team@melonport.com>
 /// @notice Trading adapter to Melon Engine
-contract EngineAdapter is DSMath, TokenUser, ExchangeAdapter {
-
-    /// @notice Buys Ether from the engine, selling MLN
-    /// @param targetExchange Address of the engine
-    /// @param orderValues [0] Min Eth to receive from the engine
-    /// @param orderValues [1] MLN quantity
-    /// @param orderValues [6] Same as orderValues[1]
-    /// @param orderAddresses [2] WETH token
-    /// @param orderAddresses [3] MLN token
+contract EngineAdapter is ExchangeAdapter, OrderFiller {
+    /// @notice Buys Ether from the Melon Engine, selling MLN
+    /// @param _targetExchange Address of the engine
+    /// @param _orderValues [0] Expected min ETH quantity (maker quantity)
+    /// @param _orderValues [1] Expected MLN quantity (taker quantity)
+    /// @param _orderValues [6] Same as orderValues[1]
+    /// @param _orderAddresses [2] WETH token (maker asset)
+    /// @param _orderAddresses [3] MLN token (taker asset)
     function takeOrder (
-        address targetExchange,
-        address[8] memory orderAddresses,
-        uint[8] memory orderValues,
-        bytes[4] memory orderData,
-        bytes32 identifier,
-        bytes memory signature
-    ) public override onlyManager notShutDown {
-        Hub hub = getHub();
+        address _targetExchange,
+        address[8] memory _orderAddresses,
+        uint[8] memory _orderValues,
+        bytes[4] memory _orderData,
+        bytes32 _identifier,
+        bytes memory _signature
+    )
+        public
```

Comment:
@travs I ran into this error while making `takeOrder` an `external` function: https://github.com/ethereum/solidity/issues/8210. So apparently we can't assign fixed bytes array calldata into memory. I'm sure this will be fixed in the future, but for now, we need to leave as `public`.

Reply:
This is actually moot now that I've moved `takeOrder` into an `OrderTaker` contract.